plugins/redfish/fu-redfish-backend.c

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright (C) 2017 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include <fwupdplugin.h>
+
+#include "fu-redfish-backend.h"
+#include "fu-redfish-common.h"
+#include "fu-redfish-device.h"
+#include "fu-redfish-request.h"
+#include "fu-redfish-smbios.h"
+
+struct _FuRedfishBackend
+{
+	FuBackend		 parent_instance;
+	gchar			*hostname;
+	gchar			*username;
+	gchar			*password;
+	guint			 port;
+	gchar			*update_uri_path;
+	gchar			*push_uri_path;
+	gboolean		 use_https;
+	gboolean		 cacheck;
+	gint64			 max_image_size;	/* bytes */
+	FuRedfishBackendKind	 kind;
+	GHashTable		*request_cache;		/* str:GByteArray */
+};
+
+G_DEFINE_TYPE (FuRedfishBackend, fu_redfish_backend, FU_TYPE_BACKEND)
+
+const gchar *
+fu_redfish_backend_kind_to_string (FuRedfishBackendKind kind)
+{
+	if (kind == FU_REDFISH_BACKEND_UPDATE_KIND_LEGACY)
+		return "mooo";
+	if (kind == FU_REDFISH_BACKEND_UPDATE_KIND_MULTIPART)
+		return "multipart";
+//	if (kind == FU_REDFISH_BACKEND_UPDATE_KIND_OEM_LENOVO)
+//		return "oem-lenovo";
```

Comment:
what's this?

Reply:
I might need to explicitly say "use this lenovo variant of Redfish for the multipart push. I'm trying to avoid it by doing fallbacks in the right place. It might be better to use a private flag for this. Not sure yet.

Reply:
Also notable is that I'm pushing the Lenovo team to align the XCC-variant of Redfish into conforming with actual redfish.

Reply:
Removed as we're pushing everything to conform to DSP2062 instead